tests/utils/dockercli/volumelifecycle.go

```diff
@@ -42,9 +42,14 @@ func CreateVolumeWithOptions(ip, name, options string) (string, error) {
 // AttachVolume - attach volume to container on given host
 func AttachVolume(ip, volName, containerName string) (string, error) {
 	log.Printf("Attaching volume [%s] on VM [%s]\n", volName, ip)
-	return ssh.InvokeCommand(ip, dockercli.RunContainer+"-d -v "+volName+
-		":/vol1 --name "+containerName+
-		dockercli.TestContainer)
+	return ssh.InvokeCommand(ip, dockercli.RunContainer+" -d --rm -v "+volName+
```

Comment:
adding --rm command will remove the container after the container exits.
We have test cases where we want the same container to start again after it exits.
We won't be able to simulate such test cases if we add this flag.

```
root@sc-rdops-vm02-dhcp-52-237:~# docker volume create -d vsphere --name test2
test2
root@sc-rdops-vm02-dhcp-52-237:~# docker run  -d --rm -v test2:/vol1 busybox tail -f /dev/null
726f2d815b18ed7f514b6d1f165033302d6c1aa36121ed2a3db4513abbf8f725
root@sc-rdops-vm02-dhcp-52-237:~# docker ps -a
CONTAINER ID        IMAGE               COMMAND               CREATED             STATUS              PORTS               NAMES
726f2d815b18        busybox             "tail -f /dev/null"   5 seconds ago       Up 3 seconds                            goofy_jennings
root@sc-rdops-vm02-dhcp-52-237:~# docker stop 726f2d815b18
726f2d815b18
root@sc-rdops-vm02-dhcp-52-237:~# docker ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
```

At the end of the above steps, I need to start the same container again, but now its already removed.

Instead, cleaning containers could be handled in the tests itself; and anyway we are about to have a common teardown that would essentially delete containers that were created as part of a test.
I agree that adding this flag would allow us to skip specifically remove containers in cleanup in some cases. But we do have a valid cause for not having this flag.

This is also the reason the restart test failed for this PR. The teardown tried to remove the container, but it was already removed due to this flag.